framework/encode/capture_settings.cpp

```diff
@@ -300,12 +190,16 @@ CaptureSettings::CaptureSettings(const TraceSettings& trace_settings)
 
 CaptureSettings::~CaptureSettings() {}
 
-void CaptureSettings::LoadSettings(CaptureSettings* settings)
+void CaptureSettings::LoadSettings(CaptureSettings* settings, const CaptureLayerSettings* layer_settings)
 {
     if (settings != nullptr)
     {
         OptionsMap capture_settings;
 
+        if (layer_settings != nullptr)
+        {
+            LoadOptionsLayer(&capture_settings, layer_settings);
+        }
```

Comment:
Instead of maintaining and populating a new settings struct, would it be possible to load everything we find in `VkLayerSettingsCreateInfoEXT` into a dumb key/value store (e.g. `OptionsMap`) and then reuse the existing processing code? This is essentially how `LoadOptionsFile` works, it just grabs whatever is in the file and lets `ProcessOptions` sort it out.